internal/helm/helm.go

```diff
@@ -333,7 +372,13 @@ func ResolveHelmChartVersion(versionFlag, chartDirectoryFlag string) (semver2.Ve
 			return semver2.Version{}, err
 		}
 		if _, err = newChartFromCiliumVersion(version); err != nil {
-			return semver2.Version{}, err
+			if !errors.Is(err, fs.ErrNotExist) {
+				return semver2.Version{}, err
+			}
+			_, err = newChartFromRemote(version)
```

Comment:
If I understand correctly, when `cilium install --version $REMOTE_CILIUM_VERSION` is executed, we will fetch the charts _twice_ (once we[ setup the `k8sInstaller`](https://github.com/cilium/cilium-cli/blob/f5b079c42bc417541f8bb7c88ddc95b171728ee0/install/install.go#L365), and again when [performing the installation](https://github.com/cilium/cilium-cli/blob/f5b079c42bc417541f8bb7c88ddc95b171728ee0/install/install.go#L619)).

I think here we should be able to do something akin to `helm search repo cilium/cilium --version $REMOTE_CILIUM_VERSION`  (i.e. validating that the version is published) without having to fetch the whole Charts. Does that make sense?